src/test/java/com/es/phoneshop/controller/ProductListPageServletTest.java

```diff
@@ -25,17 +24,21 @@
     @Mock
     private RequestDispatcher requestDispatcher;
 
-    private ProductListPageServlet servlet = new ProductListPageServlet();
+    private final ProductListPageServlet servlet = new ProductListPageServlet();
 
     @Before
-    public void setup(){
+    public void setup() throws ServletException, IOException {
         when(request.getRequestDispatcher(anyString())).thenReturn(requestDispatcher);
+        servlet.doGet(request, response);
     }
 
     @Test
-    public void testDoGet() throws ServletException, IOException {
-        servlet.doGet(request, response);
+    public void testDoGetRequestSetAttribute() {
+        verify(request).setAttribute("products", new ArrayListProductDao().findProducts());
+    }
 
-        verify(requestDispatcher).forward(request, response);
+    @Test
+    public void testDoGetForward() throws ServletException, IOException {
```

Comment:
Those two test methods are actually testing one method of the tested instance - doGet, whereas they don't really differ in terms of preconditions and workflow. It makes sense to put both those verify checks into one test (testDoGet).